src/coreclr/vm/comdelegate.cpp

```diff
@@ -1725,8 +1729,10 @@ extern "C" void QCALLTYPE Delegate_Construct(QCall::ObjectHandleOnStack _this, Q
     if (COMDelegate::NeedsWrapperDelegate(pMeth))
         refThis = COMDelegate::CreateWrapperDelegate(refThis, pMeth);
 
+    refThis->SetMethodDesc(pMethOrig);
+
     if (pMeth->GetLoaderAllocator()->IsCollectible())
-        refThis->SetMethodBase(pMeth->GetLoaderAllocator()->GetExposedObject());
+        refThis->SetInvocationList(pMeth->GetLoaderAllocator()->GetExposedObject());
```

Comment:
`InvocationList` seems to have already been used for `LoaderAllocators` in specific cases so I just moved all of them there, I'm not 100% sure if there might be a case where VM puts one there now while the field is already occupied.